bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+local goipath="${1}"
+local file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
+             "${gopath}/src/${goipath}" \
+             "${prefix}${gopath}/src/${goipath}"; do
+  [[ ${file}/ == ${proot}/* ]] && file=".${file#${proot}}"
+done
+file="$(realpath -sm ${file})"
+local workdir="$(realpath -sm .)"
+if [[ ${file}/ == ${workdir}/* ]] ; then
+  local dest="${prefix}${gopath}/src/${goipath}${file#${workdir}}"
+else
+  local dest="${prefix}${file}"
+fi
+local destdir="$(dirname ${dest})"
+if [[ ! -d "${destdir}" ]] ; then
+  installfile "${goipath}" "${destdir#${prefix}}"
+fi
+if [[ (! -e $file) || (-d "${file}" && ! -L "${file}") ]] ; then
+  install -m 0755 -vd "${dest}"
+  if [[ -d "${file}" && ! -L "${file}" ]] ; then
+    find "${file}" -iname '*.md' -type f -maxdepth 1 | while read docfile ; do
+      installfile "${goipath}" "${docfile}"
+    done
+  fi
+  local fllprefix="%dir"
+else
+  if [[ -L "${file}" ]] ; then
+    ln -s    $(readlink "${file}") "${dest}"
+    touch -h -r         "${file}"  "${dest}"
+  else
+    install -m 0644 -vp "${file}"  "${dest}"
   fi
-  [[ -f "$file" && ! -L "$file" ]] && install -m 0644 -vp  "${file}" "${destdir}/"
-  [[ -f "$file" ||   -L "$file" ]] && echo "\"${gopath}/src/${goipath}/${file}\"" >> "${filelist}" || :
+  [[ "${file}" == *.md ]] && local fllprefix="%doc"
+fi
+echo "${fllprefix:+${fllprefix} }\"${dest#${prefix}}\"" >> "${filelist}"
```

Comment:
Why do we need double quotes around the directory name?

Reply:
That's just to handle directories with spaces if we ever have to create some (though, not sure if it will work in all rpm versions, but anyway rpm does not mind the quotes so it's safer this way)